students/jerickson/Lesson5/mailroom.py

```diff
@@ -0,0 +1,294 @@
+    donor_record : dict
+        The summary info of a donor's data
+    amount : float|str
+        The amount of a new donation, will be called out in the email separately
+
+    Returns
+    -------
+    email : str
+        The composed email
+    """
+    donor_record = donation_data[donor_name]
+    time_s = "times" if donor_record["numGifts"] > 1 else "time"
+    email = f"Thank you {donor_name} for your donation of ${amount:.2f}! You have donated {donor_record['numGifts']} {time_s} for a total of ${donor_record['totalGiven']:.2f}."
+    return email
+
+
+def compose_all_donors_emails():
+    """Write to disk thank-you emails using each donors' historical information"""
+    file_id = 0
+    for donor_name, donor_record in donation_data.items():
+        time_s = (
+            f"{donor_record['numGifts']:d} donations"
+            if donor_record["numGifts"] > 1
+            else "donation"
+        )  # Grammer correction of donation vs # donations
+        email = f"Thank you {donor_name},\n\nYour {time_s} totaling ${donor_record['totalGiven']:.2f} will help us.\n\n{'':>40}Best Regards,\n{'':>40}Jacob Erickson"
+        file_name = f"Donor{file_id:03d}_{donor_name}_gitnore.txt"
+
+        path = os.path.dirname(os.path.realpath(__file__))
+
+        with open(path + "\\" + file_name, "w") as file:
+            file.write(email)
+        file_id += 1
+
+
+def thank_you():
+    """
+    Add new donation to the record and compose a thank you email to that donor
+
+    User input of donor-name and donor-amount, not case-sensitive
+        donor-name input: 'list' will show all existing donors
+        donor-name input: 'quit' will exit to the main interface and cancel the donation
+        donor-amount input: 'quit' will exit to the main interface and cancel the donation
+    The donation is recorded in the donation-data structure, new donor names are added (using case on input).
+    Print the thank-you email to the donor in the terminal. Includes historical and recent donation data.
+    """
+    donor_name = None
+
+    while not donor_name:
+        donor_name = input(
+            "Who just made a donation? Full Name please, or 'list' to show existing donors. ->: "
+        )
+        if donor_name.lower() == "list":
+            print("All Donors:" + donor_list())
+            donor_name = None
+        elif donor_name.lower() == "quit":
+            return
+
+    while True:
+        try:
+            donor_amount = input("How much was the donation? ->: ")
+            donor_amount = float(donor_amount)
+            break
+        except ValueError:
+            if donor_amount == "quit":
+                return
+            print(f"Unrecognized number: {donor_amount}. Try again.")
+
+    new_donation(donor_name, donor_amount)
+    email = compose_new_donation_email(donor_name, donor_amount)
+    print(email)
+
+
+def quit_menu():
+    """Return the string "quit" to exit a menu-level"""
+    return "quit"
```

Comment:
if you use sys.exit() here then you can utilize the switch dict and not need special checks below.

Reply:
I haven't implemented it yet, but my plan is to use this as a 'quit' a sub-menu as well using the menu-selection recursively. If I don't implement that in the oo_mailroom, I'll implement the sys.exit(). Thank you.